nslsii/__init__.py

```diff
@@ -256,30 +259,37 @@ def configure_bluesky_logging(ipython):
     ----------
     ipython: InteractiveShell
         IPython InteractiveShell used to attach bluesky log handler to ipython
+    appdirs_appname: str
+        appname passed to appdirs.user_log_dir(); use the default for production,
+        set to something else for testing
 
     Returns
     -------
-    bluesky_log_file_path: str
+    bluesky_log_file_path: Path
         log file path
 
     """
     global bluesky_log_file_path
 
     if "BLUESKY_LOG_FILE" in os.environ:
+        bluesky_log_file_path = Path(os.environ["BLUESKY_LOG_FILE"])
```

Comment:
Should we validate somehow that the path passed via the environment variable is correct?

Reply:
We were not checking before :shrug: .

I guess we could check that it exists (unless the logger will create it) and it is writabel?

Reply:
Yes, I think it's best to check as least that the path is valid. It's a good point about the writable path.

Reply:
I added tests for the case of a log directory that does not exist. The result is FileNotFoundError. I think this is the behavior we want rather than creating the directory, since the most probable cause is a mis-spelled log directory (someone knew what they wanted but a transmission error occurred at the keyboard).

I also added tests for the case of an un-writeable log directory. The result is PermissionError. Again, I think this is the behavior we want.

So while it seems like the proposed checks are the responsible thing to do, I don't think we can improve on the exceptions we already get.

Reply:
I told a lie :(. The ipython exception logger raises a warning rather than an error if the specified log directory does not exist.

I am converting that warning to an exception with `warnings.catch_warnings` and `warning.simplefilter("error")`.